internal/core/src/segcore/StreamReduce.h

```diff
@@ -0,0 +1,222 @@
+            return primary_key_ < other.primary_key_;
+        }
+        return distance_ > other.distance_;
+    }
+
+    void
+    advance() {
+        offset_++;
+        if (offset_ < offset_rb_) {
+            if (search_result_ != nullptr) {
+                primary_key_ = search_result_->primary_keys_.at(offset_);
+                distance_ = search_result_->distances_.at(offset_);
+                if (search_result_->group_by_values_.has_value() &&
+                    offset_ < search_result_->group_by_values_.value().size()) {
+                    group_by_value_ =
+                        search_result_->group_by_values_.value().at(offset_);
+                }
+            } else {
+                primary_key_ = merged_result_->primary_keys_.at(offset_);
+                distance_ = merged_result_->distances_.at(offset_);
+                if (merged_result_->group_by_values_.has_value() &&
+                    offset_ < merged_result_->group_by_values_.value().size()) {
+                    group_by_value_ =
+                        merged_result_->group_by_values_.value().at(offset_);
+                }
+            }
+        } else {
+            primary_key_ = INVALID_PK;
+            distance_ = std::numeric_limits<float>::min();
+        }
+    }
+};
+
+struct StreamSearchResultPairComparator {
+    bool
+    operator()(const std::shared_ptr<StreamSearchResultPair> lhs,
+               const std::shared_ptr<StreamSearchResultPair> rhs) const {
+        return (*rhs.get()) > (*lhs.get());
+    }
+};
+
+class StreamReducerHelper {
+ public:
+    explicit StreamReducerHelper(milvus::query::Plan* plan,
+                                 int64_t* slice_nqs,
+                                 int64_t* slice_topKs,
+                                 int64_t slice_num)
+        : plan_(plan),
+          slice_nqs_(slice_nqs, slice_nqs + slice_num),
+          slice_topKs_(slice_topKs, slice_topKs + slice_num) {
+        AssertInfo(slice_nqs_.size() > 0, "empty_nqs");
+        AssertInfo(slice_nqs_.size() == slice_topKs_.size(),
+                   "unaligned slice_nqs and slice_topKs");
+        merged_search_result = std::make_unique<MergedSearchResult>();
+        merged_search_result->has_result_ = false;
+        num_slice_ = slice_nqs_.size();
+        slice_nqs_prefix_sum_.resize(num_slice_ + 1);
+        std::partial_sum(slice_nqs_.begin(),
+                         slice_nqs_.end(),
+                         slice_nqs_prefix_sum_.begin() + 1);
+        total_nq_ = slice_nqs_prefix_sum_[num_slice_];
+    }
+
+    void
+    SetSearchResultsToMerge(std::vector<SearchResult*>& search_results) {
+        search_results_to_merge_ = search_results;
+        num_segments_ = search_results_to_merge_.size();
+        AssertInfo(num_segments_ > 0, "empty search result");
+    }
+
+ public:
+    void
+    MergeReduce();
+    void*
+    SerializeMergedResult();
+
+ protected:
+    void
+    FilterSearchResults();
```

Comment:
We need to add a short comment for each fucntion
So it's easier for reviewers to understand

Reply:
Well, for most methods in this head file, the name of method has been clear enough I suppose compared to one-sentence comment